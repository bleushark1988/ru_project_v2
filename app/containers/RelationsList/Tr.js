import styled, { css } from 'styled-components';
import Td from './Td';

const Tr = styled.tr`
  ${props =>
    props.positive &&
    css`
      background: #fcfff5;
      color: #2c662d;

      & > ${Td.selector} {
        background: #fcfff5;
        color: #2c662d;
      }
    `};
`;

export default Tr;
