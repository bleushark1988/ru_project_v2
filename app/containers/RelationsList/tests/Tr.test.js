import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Tr from '../Tr';

describe('<Tr />', () => {
  it('should render an <tr> tag', () => {
    const renderedComponent = shallow(<Tr />);
    expect(renderedComponent.type()).toEqual('tr');
  });

  it('should render colorful background when positive props is enabled', () => {
    const renderedComponent = renderer.create(<Tr positive />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
    expect(renderedComponent).toHaveStyleRule('background', '#fcfff5');
  });

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Tr />);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Tr id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = shallow(<Tr attribute="test" />);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
