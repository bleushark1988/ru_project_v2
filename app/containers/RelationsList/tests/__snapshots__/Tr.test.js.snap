// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tr /> should render colorful background when positive props is enabled 1`] = `
.c0 {
  background: #fcfff5;
  color: #2c662d;
}

.c0 > {
  background: #fcfff5;
  color: #2c662d;
}

<tr
  className="c0"
/>
`;
